File: 0038-count-and-say/0038-count-and-say.java
class Solution {
    String fun(int n)
    {
        if(n == 1) return "1";
        String str = fun(n-1);
        StringBuilder sb = new StringBuilder();

        int c = 0;
        for(int i=0; i<str.length(); i++)
        {
            c++;
            if(i == str.length()-1 || str.charAt(i) != str.charAt(i+1))
            {
                sb.append(c);
                sb.append(str.charAt(i));
                c = 0;
            }
        }
        return sb.toString();
    }
    public String countAndSay(int n) {
        return fun(n);
    }
}

File: 0038-count-and-say/0038-count-and-say.py
class Solution:
    def countAndSay(self, n: int) -> str:
        if n == 1:
            return "1"
        
        prev_str = self.countAndSay(n - 1)
        count = 0
        result = []
        
        for i in range(len(prev_str)):
            count += 1
            if i == len(prev_str) - 1 or prev_str[i] != prev_str[i + 1]:
                result.append(str(count))
                result.append(prev_str[i])
                count = 0
                
        return ''.join(result)
